kura/org.eclipse.kura.camel.cloud.factory/src/org/eclipse/kura/camel/cloud/factory/internal/XmlCamelCloudService.java

```diff
@@ -41,25 +47,32 @@ public XmlCamelCloudService(final BundleContext context, final String xml, Strin
     }
 
     public void start() throws Exception {
-        this.router = new KuraRouter() {
-            protected void beforeStart(CamelContext camelContext) {
-                camelContext.getShutdownStrategy().setTimeout(5);
-                camelContext.disableJMX();
-            };
-        };
 
-        this.service = new DefaultCamelCloudService(this.router.getContext());
+        // new router
+
+        this.router = new OsgiDefaultCamelContext(this.context);
 
-        KuraCloudComponent cloudComponent = new KuraCloudComponent(this.router.getContext());
+        // new cloud service
+        
+        this.service = new DefaultCamelCloudService(this.router);
+        
+        // set up
+
+        final KuraCloudComponent cloudComponent = new KuraCloudComponent(this.router);
         cloudComponent.setCloudService(this.service);
-        this.router.getContext().addComponent("kura-cloud", cloudComponent);
+        this.router.addComponent("kura-cloud", cloudComponent);
+        
+        final RoutesDefinition routesDefinition = this.router.loadRoutesDefinition(new ByteArrayInputStream(this.xml.getBytes()));
+        this.router.addRouteDefinitions(routesDefinition.getRoutes());
 
-        final RoutesDefinition routesDefinition = this.router.getContext().loadRoutesDefinition(new ByteArrayInputStream(this.xml.getBytes()));
-        this.router.getContext().addRouteDefinitions(routesDefinition.getRoutes());
+        // start
 
-        // register
+        logger.debug("Starting router...");
+        this.router.start();
+        final ServiceStatus status = this.router.getStatus();
+        logger.debug("Starting router... {} ({}, {})", status, status == Started, this.service.isConnected());
 
```

Comment:
My Eclipse complain about this call not matching a valid Logger.debug signature.
This is not caught by Maven though.
I've fixed changing to:
```
logger.debug("Starting router... {} ({}, {})", new Object[] {status, status == Started, this.service.isConnected()});
```